crates/bevy_ecs/src/system/commands.rs

```diff
@@ -166,6 +166,21 @@ impl<'a> Commands<'a> {
         self.queue.push(SpawnBatch { bundles_iter });
     }
 
+    /// Equivalent to iterating `bundles_iter` and calling [`Self::spawn_bundle`] on each bundle,
+    /// then collecting the entities.
+    /// If the the entities are not needed, prefer [`Self::spawn_batch`].
+    #[must_use = "If the returned entities are not needed, use [`Self::spawn_batch`] instead."]
+    pub fn spawn_batch_reserved<I>(&mut self, bundles_iter: I) -> Vec<Entity>
```

Comment:
I think this should just return `impl Iterator<Item=Entity>` to avoid allocating. Users can always collect the entities on their own.